### PR TITLE
Working Travis build, with caching of deps

### PR DESCRIPTION
**Pull request by ejona86**
It will likely fail tests, due to #223.

### CODE REVIEW

File: compiler/build.gradle [hunk 1]
```diff
@@ -25,7 +25,7 @@ dependencies {
 
 binaries.all {
   if (toolChain in Gcc || toolChain in Clang) {
-    cppCompiler.args "-std=c++11"
+    cppCompiler.args "--std=c++0x"
```
Comment (louiscryan) [inline compiler/build.gradle]: maybe add a comment here about support gcc-4.6 and below

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  - linux
+
+branches:
+  only:
+    - master
+
+notifications:
+  email: false
+
+cache:
```
Comment (louiscryan) [inline .travis.yml]: super nifty!

**Comment by louiscryan** [comment]
LGTM. Getting the script working on OSX would be the next step I guess

**Comment by ejona86** [comment]
Okay. I'll commit it soon. I'm checking to see if the gradle cache actually works. I'm worried about changed files (locks, stray modifications) from invalidating the cache. If it does, then I'll remove it before merging.

**Comment by ejona86** [comment]
I've dropped gradle caching. I've got pieces of it working, but I'll need some more time to do some trials. Little sense in delaying this.

**Pull request merged**